redis-om-spring/src/main/java/com/redis/om/spring/metamodel/MetamodelGenerator.java

```diff
@@ -296,6 +311,61 @@ else if (Set.class.isAssignableFrom(targetCls) || List.class.isAssignableFrom(ta
     writer.close();
   }
 
+  private List<Pair<FieldSpec, CodeBlock>> extractNestedConstants(Element fieldElement) {
+    TypeMirror typeMirror = fieldElement.asType();
+    DeclaredType asDeclaredType = (DeclaredType)typeMirror;
+    Element entityField = asDeclaredType.asElement();
+
+    messager.printMessage(Diagnostic.Kind.NOTE, "Processing constants for " + fieldElement + " of type " + entityField);
+
+    final String entityFieldName = fieldElement.toString();
+    messager.printMessage(Diagnostic.Kind.NOTE, "entityFieldName => " + entityFieldName);
+
+    List<? extends Element> enclosed = entityField.getEnclosedElements();
+    messager.printMessage(Diagnostic.Kind.NOTE, "enclosed size() ==> " + enclosed.size());
+
+
+    final Map<String, Element> getters = entityField.getEnclosedElements().stream()
+        .filter(ee -> ee.getKind() == ElementKind.METHOD)
+        // Only consider methods with no parameters
+        .filter(ee -> ee.getEnclosedElements().stream().noneMatch(eee -> eee.getKind() == ElementKind.PARAMETER))
+        // Todo: Filter out methods that returns void or Void
+        .collect(Collectors.toMap(e -> e.getSimpleName().toString(), Function.identity()));
+
+    messager.printMessage(Diagnostic.Kind.NOTE, "getters size() ==> " + getters.size());
+
+    final Set<String> isGetters = getters.values().stream()
+        // todo: Filter out methods only returning boolean or Boolean
```

Comment:
TODO?

Reply:
I'll open a ticket, I'm waiting for the lib upstream to implement some utilities so that I don't have to do a mess with reflection